src/components/Card.test.js

```diff
@@ -0,0 +1,40 @@
+import { render, screen } from "@testing-library/react";
+import { waitFor } from "@testing-library/react";
+
+import userEvent from "@testing-library/user-event";
+import Card from "./Card";
+
+describe("Card", () => {
+	test("should render input", () => {
+		render(<Card />);
+		const input = screen.getByText("Card number");
+		expect(input).toBeInTheDocument();
+	});
+	test("should render button", () => {
+		render(<Card />);
+		const submitButton = screen.getByRole("button");
+		expect(submitButton).toBeInTheDocument();
+	});
+
+	test("should return: Your card is not valid if card number is invalid", async () => {
+		render(<Card />);
+		const cardNumber = "0000011";
+		const input = screen.getByText("Card number");
+		const submitButton = screen.getByRole("button");
+		userEvent.type(input, cardNumber);
+		userEvent.click(submitButton);
+		const paragraph = await screen.findByText("Your card is not valid");
+		expect(paragraph.textContent).toBe("Your card is not valid");
+	});
+
+	test("should return 'Visa' when first digit is 4 and there are 16 digits in total", async () => {
```

Comment:
Ten test nie przechodzi, wpisując ten numer na stronie też mamy informację: `Your card is not valid`.
To oznacza, że albo jest to zły numer, albo źle walidacja została napisana ;)

Idąc tym tropem zorientowałem się, że funkcja `checkCardName` nie zostaje w ogóle uruchomiona!
Okazuje się, że `isCardNumberValid` zwraca `false` i na tym kończy się działanie.
Potwierdzam - zły numer karty wpisałaś :P

PS. Okazuje się, że tak funkcja `isCardNumberValid` nie działa poprawnie - teoretycznie prawidłowy numer też nieprzechodzi ;)